### ISSUE TITLE
"the last matching pattern takes the most precedence" is not followed

### ISSUE THREAD

**Issue by sbergen** [dounan/codeowners-chrome#9]
Referring to the documentation at https://help.github.com/articles/about-codeowners/ it seems to me that the plugin does not follow this rule.

Unfortunately, I couldn't figure out a really simple fix for this, as ultimately the logic of gathering all the lines applying to the *current user*, and then matching files against this is not sufficient to produce the right behavior.

As an example of what will fail is the following entries:
```
TopLevelDir/ @loggedinuser
TopLevelDir/NestedDir/ @notloggedinuser
```

Here anything in `TopLevelDir/NestedDir/` will still be marked as owned by @loggedinuser, even though the specification says otherwise.

We've developed a utility we use in a C# project, where the code to find the correct owner looks like this, if it's of any help
`var entry = GetEntries().LastOrDefault(e => e.Matches(path));`
Where `GetEntries()` returns **all** the owner entries, and `Matches` would be similar to `ignores`.
Only after getting the correct entry, should the check be done to see if the current user is part of it.

**Comment by sbergen** [comment]
As GitHub now shows code owners natively, this is no longer relevant.
